prow/cmd/webhook-server/main.go

```diff
@@ -18,148 +18,176 @@ package main
+		return err
+	}
+	if o.expiryInYears < 0 {
+		return fmt.Errorf("invalid expiry years")
+	}
+	if len(o.projectId) > 0 && len(o.fileSystemPath) > 0 {
+		return fmt.Errorf("both projectid and filesystem path cannot be specified")
+	}
+	if len(o.projectId) == 0 && len(o.fileSystemPath) == 0 {
+		return fmt.Errorf("either projectid or filesystem path must be specified")
+	}
+	return nil
+}
+
+func gatherOptions(fs *flag.FlagSet, args ...string) options {
+	var o options
+	fs.StringVar(&o.projectId, "project-id", "", "Project ID for storing GCP Secrets")
+	fs.StringVar(&o.fileSystemPath, "filesys-path", "", "File system path for storing ca-cert secrets")
+	fs.IntVar(&o.expiryInYears, "expiry-years", 30, "CA certificate expiry in years")
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Whether to mutate any real-world state")
+	optionGroup := []flagutil.OptionGroup{&o.kubernetes}
+	optionGroup[0].AddFlags(fs)
+	fs.Parse(args)
+	return o
 }
 
 func main() {
 	logrusutil.ComponentInit()
 	logrus.SetLevel(logrus.DebugLevel)
-	p := flag.String("project-id", 
-	"", "Project ID for storing GCP Secrets")
-	e := flag.Int("expiry-years", 30, "CA certificate expiry in years")
-	dns := flagutil.NewStrings("validation-webhook-service", "validation-webhook-service.default", "validation-webhook-service.default.svc")
+	dns := prowflagutil.NewStrings("validation-webhook-service", "validation-webhook-service.default", "validation-webhook-service.default.svc")
 	flag.Var(&dns, "dns", "DNS Names CA-Cert config")
 	flag.Parse()
-	exp := *e
-	projectId := *p
-	if len(projectId) == 0 {
-		logrus.Fatal("project-id flag not supplied")
+	o := gatherOptions(flag.NewFlagSet(os.Args[0], flag.ExitOnError), os.Args[1:]...)
+	o.dnsNames = dns.Strings()
+	if err := o.Validate(); err != nil {
+		logrus.WithError(err).Fatal("Invalid options")
 	}
-	client, err := secretmanager.NewClient(projectId, false)
+	kubeCfg, err := o.kubernetes.InfrastructureClusterConfig(o.dryRun)
 	if err != nil {
-		logrus.WithError(err).Fatal("Unable to create secret manager client")
+		logrus.WithError(err).Fatal("Error getting kubeconfig")
 	}
+	var certFile string
+	var privKeyFile string
+	var tempDir string
 	ctx := context.Background()
-	cert, privKey, err := getGCPSecrets(client, ctx, exp, dns)
-	if err != nil && strings.Contains(err.Error(), gcpSecretError) {
-		logrus.Infof("%v, Will now proceed to create GCP Secret", err)
-		cert, privKey, err = createGCPSecret(client, ctx, exp, dns.Strings())
+	cl, err := ctrlruntimeclient.New(kubeCfg, ctrlruntimeclient.Options{})
+	if err != nil {
+		logrus.WithError(err).Fatal("Could not create writer client")
+	}
+	var client ClientInterface
+	if len(o.projectId) > 0 {
+		secretManagerClient, err := secretmanager.NewClient(o.projectId, false)
 		if err != nil {
-			logrus.WithError(err).Fatal("Unable to create ca certificate")
+			logrus.WithError(err).Fatal("Unable to create secretmanager client", err)
+		}
+		client := newGCPClient(secretManagerClient)
```

Comment:
This defines a new `client` variable at this scope rather than setting the variable defined on line 114.